lib/gitlab/ci/templates/pipeline/AndroidTemplate.yml

```diff
@@ -129,6 +129,22 @@ testRelease:
 instrumentationTests:
   rules: *debug_rules
 
+# Save report from secret_detection job
+secret_detection:
+  artifacts:
+    reports:
+      secret_detection: gl-secret-detection-report.json
+    paths:
+      - gl-secret-detection-report.json
+
+# Save report from semgrep-sast scan job
+semgrep-sast:
```

Comment:
Looking at the default SAST job rules at https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/gitlab/ci/templates/Jobs/SAST.gitlab-ci.yml#L172 it doesn't include merge request pipelines.

Should we update the `semgrep-sast` job's rules for all these pipeline templates to include merge request pipelines so we can detect SAST findings during merge requests? Would be something like the below. 

```
  # Note, overrides the default rules at https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/gitlab/ci/templates/Jobs/SAST.gitlab-ci.yml. Makes some assumptions like "exists" for language files types like ".py" not needed because we are explicitly deciding to turn on semgrep-sast for a given language Gitlab pipeline. Also Gitlab Advanced SAST checks not done since that is a Gitlab Ultimate feature we're going to ignore since not many/any users using Gitlab Ultimate. 
  rules:
   - if: $SAST_DISABLED == 'true' || $SAST_DISABLED == '1'
      when: never
    - if: $SAST_EXCLUDED_ANALYZERS =~ /semgrep/
      when: never
    - if: $CI_COMMIT_BRANCH
    - if: $CI_MERGE_REQUEST_IID
 ```